lib/galaxy/datatypes/sequence.py

```diff
@@ -795,16 +796,36 @@ def quality_check(lines):
     @classmethod
     def check_first_block(cls, file_prefix):
         # check that first block looks like a fastq block
-        headers = get_headers(file_prefix, sep='\n', count=4)
-        if len(headers) == 4 and headers[0][0] and headers[0][0][0] == "@" and headers[2][0] and headers[2][0][0] == "+" and headers[1][0]:
+        block = get_headers(file_prefix, sep='\n', count=4)
+        return cls.check_block(block)
+
+    @classmethod
+    def check_block(cls, block):
+        if len(block) == 4 and block[0][0] and block[0][0][0] == "@" and block[2][0] and block[2][0][0] == "+" and block[1][0]:
             # Check the sequence line, make sure it contains only G/C/A/T/N
-            match = cls.bases_regexp.match(headers[1][0])
+            match = cls.bases_regexp.match(block[1][0])
             if match:
                 start, end = match.span()
-                if (end - start) == len(headers[1][0]):
+                if (end - start) == len(block[1][0]):
                     return True
         return False
 
+    def validate(self, dataset, **kwd):
+        headers = iter_headers(dataset.file_name, sep='\n', count=-1)
+        # check to see if the base qualities match
```

Comment:
Can we add here a TODO ... this is currently not implemented, isn't it?
We will work on this in the next release cycle.

Reply:
This was a copy and pasted comment, but isn't this what quality_check does in the nextline?

Reply:
The function is defined as ...
```
def quality_check(lines):
        return True
```

maybe I miss something.

Reply:
The subclasses implement it I thought - at least FastqSanger does:

```python
    @staticmethod
    def quality_check(lines):
        """Presuming lines are lines from a fastq file, return True if the qualities are compatible with sanger encoding"""
        for line in islice(lines, 3, None, 4):
            if not all(_ >= '!' and _ <= 'S' for _ in line[0]):
                return False
        return True
```

Looks like we could definitely do more validation for sure though.